src/client/cli/cmd/aliases.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2021 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "aliases.h"
+#include "common_cli.h"
+
+#include <multipass/cli/argparser.h>
+
+namespace mp = multipass;
+namespace cmd = multipass::cmd;
+
+mp::ReturnCode cmd::Aliases::run(mp::ArgParser* parser)
+{
+    auto ret = parse_args(parser);
+    if (ret != ParseCode::Ok)
+    {
+        return parser->returnCodeFrom(ret);
+    }
+
+    cout << formatter.format(aliases);
+
+    return ReturnCode::Ok;
+}
+
+std::string cmd::Aliases::name() const
+{
+    return "aliases";
+}
+
+QString cmd::Aliases::short_help() const
+{
+    return QStringLiteral("List available aliases");
+}
+
+QString cmd::Aliases::description() const
+{
+    return QStringLiteral("List available aliases");
+}
+
+mp::ParseCode cmd::Aliases::set_formatter(mp::ArgParser* parser)
+{
+    std::string user_arg = parser->value(format_option_name).toStdString();
+
+    if (user_arg != "csv" && user_arg != "json" && user_arg != "table" && user_arg != "yaml")
+    {
+        fmt::print(cerr, "Invalid format type given.\n");
+        return ParseCode::CommandLineError;
+    }
```

Comment:
I'm not sure if you know this, but there is already a function in client/cli/cmd/common_cli.cpp called handle_format_option() that basically does all of this.

Reply:
Ah, right, it will return a different type.  But yeah, if it's not *too* much work to factor something out that can be shared between the two, that would be great.

Reply:
I did the refactor, finally. Code looks *much* better now, thanks for the suggestion!